openwhisk/openwhisk_apigateway.md

```diff
@@ -15,58 +15,111 @@ lastupdated: "2017-03-16"
-```
-{
-  "marco":"polo"
-}
-```
-{: screen}
-
-You can pass parameters to the action via simple query parameters, or via request body.
-
-### Exposing multiple actions
-{: #openwhisk_apigateway_actions}
+Configure your OpenWhisk with the apihost `wsk property set --apihost openwhisk.ng.bluemix.net`
+To be able to use the `wsk api` the CLI configuration file `~/.wskprops` needs to contain the Bluemix Access Token.
+To get the access token use the CLI command `wsk bluemix login`, for more information about the command run `wsk bluemix login -h`
+
+Note: If the command errors requiring single sign on (sso), this is not currently supported. As a workaround login with the Bluemix CLI using `bluemix login`, then copy the Access Token from the HOME directory configuration file `~/.bluemix/config.json` into the `~/.wskprops` file as the property `APIGW_ACCESS_TOKEN="value of IAMToken`. Remove the prefix `Bearer ` when copying the access token string.
+
+## Expose an OpenWhisk web action
+
+1. Create a JavaScript file with the following content. For this example, the file name is 'hello.js'.
+  ```javascript
+  function main({name:name='Serverless API'}) {
+      return {payload: `Hello world ${name}`};
+  }
+  ```
+  {: codeblock}
+  
+2. Create a web action from the following JavaScript function. For this example, the action is called 'hello'. Make sure to add the flag `--web true`
+  
+  ```
+  wsk action update hello hello.js --web true
+  ```
+  {: pre}
+  ```
+  ok: updated action hello
+  ```
+  
+3. Create an API with base path `/hello`, path `/world` and method `get` with response type `json`
+  
+  ```
+  wsk api create /hello /world get hello --response-type json
+  ```
+  {: pre}
+  ```
+  ok: created API /hello/world GET for action /_/hello
+  https://service.us.apiconnect.ibmcloud.com/gws/apigateway/api/21ef035/hello/world
+  ```
+  A new URL is generated exposing the `hello` action via a __GET__ HTTP method.
+  
+4. Let's give it a try by sending a HTTP request to the URL.
+  
+  ```
+  $ curl https://service.us.apiconnect.ibmcloud.com/gws/apigateway/api/21ef035/hello/world?name=OpenWhisk
+  ```
+  {: pre}
+  ```json
+  {
+  "payload": "Hello world OpenWhisk"
+  }
+  ```
+  The action `hello` got invoked, returning back a JSON string including the parameter `name` sent via query parameter. You can pass parameters to the action via simple query parameters, or via request body.
+  
+5. Full controll over the the http response
+  
+  Notice that the full result of the action is returned in the body of the response, and the content is assume to be `application/json` this is because the API was created with `--response-type json` using `json` for the response-type allows you to easily get started with existing actions, but once you have something working you want to have full control over the http response like `statusCode`, `headers` and return different content types in the `body`.
```

Comment:
Run-on sentence.  Fix by adding a period.  `application/json. This is because the API was...`

Add a `which`
-> `...for the response-type, which allows you to easily...`

Add `may`
```...working you may want to have full control...```